src/pages/vendorpages/Product.tsx

```diff
@@ -0,0 +1,176 @@
+import { ActionButton } from '../../components/vendorcomponents/ActionButton';
+
+const Product: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const productsPerPage: number = 8; // Number of products to display per page
+  const { data, isLoading, error } = useFetch('/api/v1/products/all');
+
+  if (error) return <ErrorPage message="Error while connecting to server" />;
+  if (isLoading) return <Loading message="Loading products..." />;
+
+  // Filtered products based on search term
+  const filteredProducts: IProduct[] = data.data.filter((product: IProduct) =>
+    product.name.toLowerCase().includes(searchTerm.toLowerCase()),
+  );
+  const formatDate = (dateString: string): string => {
+    const date = new Date(dateString);
+    return date.toLocaleDateString(); // Adjust options as per your preference
+  };
+  // Logic to paginate products
+  const indexOfLastProduct: number = currentPage * productsPerPage;
+  const indexOfFirstProduct: number = indexOfLastProduct - productsPerPage;
+  const currentProducts: IProduct[] = filteredProducts.slice(
+    indexOfFirstProduct,
+    indexOfLastProduct,
+  );
+
+  // Handle pagination click
+  const paginate = (pageNumber: number) => setCurrentPage(pageNumber);
+
+  return (
+    <main className="w-[96%] mx-auto">
+      <div className="flex justify-between my-2 mx-2 mb-4">
+        <h1 className="text-2xl font-bold">Product</h1>
+        <form className="flex-1 max-w-md">
+          <div className="relative">
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="24"
+              height="24"
+              viewBox="0 0 24 24"
+              fill="none"
+              stroke="currentColor"
+              strokeWidth="2"
+              strokeLinecap="round"
+              strokeLinejoin="round"
+              className="absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground"
+            >
+              <circle cx="11" cy="11" r="8"></circle>
+              <path d="m21 21-4.3-4.3"></path>
+            </svg>
+            <input
+              className="flex h-10 rounded-md border border-input bg-background px-3 py-1 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 pl-8 w-full"
+              placeholder="Search..."
+              type="search"
+              value={searchTerm}
+              onChange={(e) => setSearchTerm(e.target.value)}
+            />
+          </div>
+        </form>
+        <ActionButton
+          text="Add product"
+          colorClasses="bg-slate-800 text-white"
+          IconComponent={FaPlus}
+        />
+      </div>
+      <main>
+        <table className="min-w-full border bg-white">
```

Comment:
Create a folder called tables under components and from there add Table component which will call TableHead, TableBody, etc...